From 5695b49c7dc26a9667df223672ed221ab767a3c3 Mon Sep 17 00:00:00 2001
From: Ola Olsson <olax.olsson@intel.com>
Date: Mon, 7 Nov 2011 10:42:26 -0800
Subject: [PORT FROM MAIN] libcorkscrew: Fix hang during backtrace

BZ: 13909

Skip lookups of addresses if the symbol table is not found during
tombstone generation

Change-Id: Ic06cc9c6cd0ad854aa62bbeba82b3688b4d39e5e
Orig-Change-Id: I6d7ede5d46a92d0180bdbc0f2b6d2ef405e5c2db
Signed-off-by: Bruce Beare <bruce.j.beare@intel.com>
Reviewed-on: http://android.intel.com:8080/73415
Reviewed-by: lab_cactus <lab_cactus@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Chen, HuaqiangX <huaqiangx.chen@intel.com>
Tested-by: Rocher, Jeremy <jeremy.rocher@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Reviewed-on: http://android.intel.com:8080/78856
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 libcorkscrew/symbol_table.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libcorkscrew/symbol_table.c b/libcorkscrew/symbol_table.c
index 29e4a79..14c1089 100644
--- a/libcorkscrew/symbol_table.c
+++ b/libcorkscrew/symbol_table.c
@@ -91,7 +91,7 @@ symbol_table_t* load_symbol_table(const char *filename) {
             dynsym_idx = i;
         }
     }
-    if (dynsym_idx == -1 && sym_idx == -1) {
+    if (dynsym_idx == -1 || sym_idx == -1) {
         goto out_unmap;
     }
 
-- 
1.7.4.1

